api/src/routes/certificate.test.ts

```diff
@@ -0,0 +1,307 @@
+import { Certification } from '../../../shared/config/certification-settings';
+import {
+  defaultUserEmail,
+  defaultUserId,
+  devLogin,
+  setupServer,
+  superRequest
+} from '../../jest.utils';
+
+describe('certificate routes', () => {
+  setupServer();
+  describe('Authenticated user', () => {
+    let setCookies: string[];
+
+    // Authenticate user
+    beforeAll(async () => {
+      setCookies = await devLogin();
+    });
+
+    describe('PUT /certificate/verify', () => {
+      beforeEach(async () => {
+        await fastifyTestInstance.prisma.user.updateMany({
+          where: { email: defaultUserEmail },
+          data: {
+            completedChallenges: [],
+            name: 'fcc',
+            isRespWebDesignCert: false,
+            isJsAlgoDataStructCert: false,
+            isFrontEndLibsCert: false,
+            is2018DataVisCert: false,
+            isRelationalDatabaseCertV8: false,
+            isApisMicroservicesCert: false,
+            isQaCertV7: false,
+            isSciCompPyCertV7: false,
+            isDataAnalysisPyCertV7: false,
+            isInfosecCertV7: false,
+            isMachineLearningPyCertV7: false,
+            isCollegeAlgebraPyCertV8: false,
+            isFoundationalCSharpCertV8: false,
+            username: 'fcc'
+          }
+        });
+      });
+
+      test('should return 500 if no certSlug', async () => {
+        const response = await superRequest('/certificate/verify', {
+          method: 'PUT',
+          setCookies
+        }).send({ certSlug: undefined });
+        expect(JSON.parse(response.text)).toMatchObject({
+          code: 'FST_ERR_FAILED_ERROR_SERIALIZATION'
+        });
+        expect(response.status).toBe(500);
+      });
+
+      test('should return 400 if certSlug is invalid', async () => {
+        const response = await superRequest('/certificate/verify', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          certSlug: 'non-existant'
+        });
+        expect(response.status).toBe(400);
+      });
+
+      test('should return 500 if user not found in db', async () => {
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        const temp = fastifyTestInstance.prisma.user.findUnique;
```

Comment:
I recommend using either `jest.spyOn` or `jest.replaceProperty`, here. Probably the latter, since you don't seem to be testing how the method is called.

That means you can use `jest.restoreAllMocks` in `afterEach` rather than having to restore it inside the test. It keeps the test more focused on what it's trying to do.